packages/contracts/contracts/L1/Colosseum.sol

```diff
@@ -577,20 +671,21 @@ contract Colosseum is Initializable, ISemver {
         address _asserter,
         bytes32 _outputRoot,
         bytes32 _publicInputHash
-    ) external onlySecurityCouncil {
-        require(
-            _outputRoot != DELETED_OUTPUT_ROOT,
-            "Colosseum: cannot rollback output to zero hash"
-        );
-        require(
-            L2_ORACLE.getL2Output(_outputIndex).outputRoot == DELETED_OUTPUT_ROOT,
-            "Colosseum: only can rollback if the output has been deleted"
-        );
+    ) external onlySecurityCouncil outputNotFinalized(_outputIndex) {
+        if (_outputRoot == DELETED_OUTPUT_ROOT) revert CannotRollbackOutputToZero();
+        if (L2_ORACLE.getL2Output(_outputIndex).outputRoot != DELETED_OUTPUT_ROOT)
+            revert OutputNotDeleted();
         verifiedPublicInputs[_publicInputHash] = false;
 
         // Rollback output root.
         L2_ORACLE.replaceL2Output(_outputIndex, _outputRoot, _asserter);
 
+        // Switch validator system after validator pool contract terminated.
+        if (L2_ORACLE.VALIDATOR_POOL().isTerminated(_outputIndex)) {
+            // Unjail asserter.
+            L2_ORACLE.VALIDATOR_MANAGER().tryUnjail(_asserter, true);
+        }
+
```

Comment:
Just a question, why is this part added?

Reply:
It's because when malicious challenger has done `proveFault` and asserter has been jailed, security council should unjail the asserter when `dismissChallenge`.